lib/xnvme_be_macos_driverkit_admin.cpp

```diff
@@ -0,0 +1,63 @@
+// SPDX-FileCopyrightText: Samsung Electronics Co., Ltd
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <map>
+extern "C" {
+
+#include <libxnvme.h>
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_MACOS_ENABLED
+#include <mach/mach_error.h>
+#include <CoreFoundation/CoreFoundation.h>
+
+#include <xnvme_dev.h>
+#include <xnvme_be_macos_driverkit.h>
+
+std::map<uint64_t, std::tuple<uint64_t, uint64_t>> address_mapping;
```

Comment:
I don't like this global variable, it seems like the kind of thing that should be inside the `xnvme_be_macos_driverkit_state`

Reply:
I think I did it because it's a C++ dependency. And `struct xnvme_be_macos_driverkit_state` is used from both cpp and c files.

I've made a workaround now.

Reply:
That's better in my opinion